cpp/harvest/marc_issn_lookup.cc

```diff
@@ -106,79 +115,91 @@ void UpdateSubfield(MARC::Subfields &subfields, const SubFieldInfo &sub_field_in
 }
 
 
-void UpdateJournalValidity(std::map<std::string, SubFieldInfo> &journal_cache) {
-    // check whether the issn is occur more than once
-    unsigned online_valid(0), online_invalid(0), print_valid(0), print_invalid(0);
-
-    LOG_INFO("Cache contents:");
-    for (auto sfi : journal_cache) {
-        if (sfi.second.online_version_counter_ == 1) {
-            journal_cache[sfi.first].is_valid_ = true;
-            LOG_INFO(sfi.first + ": Online (valid)");
-            ++online_valid;
-        } else if (sfi.second.online_version_counter_ > 1) {
-            journal_cache[sfi.first].is_valid_ = false;
-            LOG_INFO(sfi.first + ": Online (invalid)");
-            ++online_invalid;
-        } else if (sfi.second.online_version_counter_ == 0)
-            if (sfi.second.printed_version_counter_ == 1) {
-                journal_cache[sfi.first].is_valid_ = true;
-                LOG_INFO(sfi.first + ": Print (valid)");
-                ++print_valid;
-            } else {
-                journal_cache[sfi.first].is_valid_ = false;
-                LOG_INFO(sfi.first + ": Print (invalid)");
-                ++print_invalid;
-            }
+void UpdateSubFieldInfo(SubFieldInfo &sfi, const SubFieldInfo &new_sfi, const bool is_online) {
+    sfi.t_ = new_sfi.t_;
+    sfi.w_ = new_sfi.w_;
+    sfi.x_ = new_sfi.x_;
+    sfi.is_online_ = is_online;
+    (is_online ? ++sfi.online_version_counter_ : ++sfi.printed_version_counter_);
+    // sfi.issns.emplace_back(new_sfi.issns);
+    sfi.issns.insert(sfi.issns.end(), new_sfi.issns.begin(), new_sfi.issns.end());
+}
+
+
+bool IsInISSNs(const std::vector<std::string> &issns, const std::vector<std::string> &issns_input) {
+    for (const auto &issn : issns_input) {
+        if (std::find(issns.begin(), issns.end(), StringUtil::ASCIIToUpper(issn)) != issns.end())
+            return true;
+    }
+    return false;
+}
+
+
+// avoiding duplication in issns's cache
+void InsertIfNotExist(const std::vector<std::string> &issns_input, std::vector<std::string> &issns) {
```

Comment:
Actually, our convention is: If a variable is modified inside a function, we always pass a pointer instead of a reference. So the second argument should be:
`std::vector<std::string> &issns` => `std::vector<std::string> * const issns`

This also increases the code readability at the points where the function is called.